daffodil-japi/src/main/scala/org/apache/daffodil/japi/schema/RuntimeSchemaWalker.scala

```diff
@@ -0,0 +1,135 @@
+package org.apache.daffodil.japi.schema
+
+import org.apache.daffodil.japi.{ DataProcessor => JDataProcessor }
+import org.apache.daffodil.lib.exceptions.Assert
+import org.apache.daffodil.runtime1.dpath.NodeInfo.PrimType
+import org.apache.daffodil.runtime1.processors.{
+  ChoiceRuntimeData,
+  DataProcessor => SDataProcessor,
+  ElementRuntimeData,
+  ErrorERD,
+  SequenceRuntimeData,
+  TermRuntimeData,
+}
+
+/**
+ * Base class used by japi clients who want to walk the runtime schema information.
+ */
+abstract class RuntimeSchemaHandler() {
```

Comment:
Name: make clear this is walking the schema, but not the DSOM schema, it's walking the schema as the Runtime1 sees it, which is the metadata needed to interface Runtime1 with other data frameworks. It's the metadata of the data that will be presented to that framework. 

"RuntimeSchema" doesn't seem sufficient.

Reply:
We already have a DSOM walker used by NiFi, is that not sufficient? Can you add comments about how the two differ?